netlogo-gui/src/main/lab/gui/Supervisor.scala

```diff
@@ -74,21 +75,31 @@ class Supervisor(dialog: java.awt.Dialog,
       val fileName = org.nlogo.swing.FileDialog.showFiles(
         workspace.getFrame, "Exporting as spreadsheet", java.awt.FileDialog.SAVE,
         workspace.guessExportName(worker.protocol.name + "-spreadsheet.csv"))
-      addExporter(new SpreadsheetExporter(
+      try{
+	    addExporter(new SpreadsheetExporter(
         workspace.getModelFileName,
         workspace.world.getDimensions,
         worker.protocol,
-        new java.io.PrintWriter(new java.io.FileWriter(fileName))))
+		new java.io.PrintWriter(new java.io.FileWriter(fileName))))
+	  }
+	  catch{
+		case e: IOException => failure(e); return
```

Comment:
I think I see why the `; return` is here as this is copied from above.  If you're going to split things onto separate lines I'd prefer to drop the `;` and have the `failure(e)` and `return` on their own lines.  You could also do things are they are above in this file and keep it all on a single line, `catch { case e: IOException => failure(e); return }`